File: AppSuViet(backendAPI)/Models/QuizQuestion.cs
namespace DACN.Models
{
    public class QuizQuestion
    {
        public int Id { get; set; }

        // Giai đoạn – để API lọc
        public string Era { get; set; } = string.Empty;
        // Ví dụ: "1856-1930"

        public string Question { get; set; } = string.Empty;

        // Danh sách 4 lựa chọn
        public List<string> Options { get; set; } = new();

        // Chỉ số đáp án đúng (0–3)
        public int CorrectIndex { get; set; }

        // Giải thích sau khi trả lời
        public string Explanation { get; set; } = string.Empty;
    }

}

File: AppSuViet(FlutterDart)/lib/models/quiz_question.dart
class QuizQuestion {
  final int id;
  final String era;
  final String question;
  final List<String> options;
  final int correctIndex;
  final String explanation;

  QuizQuestion({
    required this.id,
    required this.era,
    required this.question,
    required this.options,
    required this.correctIndex,
    required this.explanation,
  });

  factory QuizQuestion.fromJson(Map<String, dynamic> json) {
    return QuizQuestion(
      id: json['id'],
      era: json['era'],
      question: json['question'],
      options: List<String>.from(json['options']),
      correctIndex: json['correctIndex'],
      explanation: json['explanation'],
    );
  }
}
